packages/components/src/Text/Text.tsx

```diff
@@ -2,6 +2,12 @@ import React, { PropsWithChildren } from "react";
 import { Typography, TypographyOptions, TypographyProps } from "../Typography";
 
 export interface TextProps {
+  /**
+   * The HTML element to render the text as.
+   * @default "p"
+   */
+  readonly element?: TypographyProps["element"];
```

Comment:
One minor comment, is the I think you should [extract](https://www.typescriptlang.org/docs/handbook/utility-types.html#extracttype-union) all elements but `h<x>` or [exclude](https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers) the `h<x>` elements to keep text as text and headings as heading

Reply:
Great call out - done here 43bcf2e65591b99369586edd0890ed5a0ff559bb
I decided to extract everything except for headings to maintain that clear separation 🫡